### PR TITLE
feature/delete ft category implementation

### PR DESCRIPTION
**Pull request by remkro**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java [hunk 1]
```diff
@@ -9,6 +9,8 @@ public enum ErrorCode {
 
     FT001("FT001", "FINANCIAL_TRANSACTION_NOT_FOUND", 404),
 
+    FTC001("FTC001", "FINANCIAL_TRANSACTION_CATEGORY_NOT_FOUND", 404),
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java:12]: The value of the error code is
`FINANCIAL TRANSACTION CATEGORY NOT FOUND`

But in `FinancialTransactionCategoryServiceImpl` class error message is:
```java
String.format("Financial transaction category with given id: %d does not exist", id));
```

Should the convention be the same?
- not found
or
- not exist?

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java (hunk 1 above)
Comment (remkro) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java:12]: I was suggested by that line:

`String.format("FinancialTransaction with given id: %d does not exist", id));`

What's the final convention?

Comment (dawciobiel) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java:12]: > I was suggested by that line:
> 
> `String.format("FinancialTransaction with given id: %d does not exist", id));`
> 
> What's the final convention?

Check how it is now in the code:

1. Search the project for occurrences of words:
a) "String.format"
b) "not exist"
c) "not found"
Then you'll see if the convention repeats itself.

2. Check if there is, and if so, what convention is inside the enum ErrorCode.

Then you decide by yourself. And if there is no convention anywhere, then it's worth raising the topic on the #os-backend channel. To do so, create a thread under this channel.
The advantage of such a solution in the worst case will be submitting the task for later naming refactoring.

**Review by Krisu93 — DISMISSED** [review]
Looks good for me

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 2]
```diff
@@ -41,4 +45,15 @@ public List<FinancialTransactionCategoryDTO> getFinancialTransactionCategories()
                 .toList();
     }
 
+    @Override
+    public void deleteFinancialTransactionCategory(@Min(1) @NotNull Long id) {
+        if (financialTransactionCategoryRepository.existsById(id)) {
+            financialTransactionCategoryRepository.deleteById(id);
```
[the remark this answers is not in the document]
Comment (dawciobiel) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:51]: It's going to be implemented in another issue in the future.

**Review by dawciobiel — DISMISSED** [review]
Looks good to me.

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Comment by dawciobiel** [comment]
This pull request going to be re-created.

**Pull request closed without merging**